src/services/identity/SitesService.ts

```diff
@@ -175,27 +189,49 @@ class SitesService {
     }
 
     // Legacy style of commits, or if the user is not found
-    const {
-      author: { email: authorEmail },
-    } = commit
-    return authorEmail
+    return this.extractAuthorEmail(commit)
   }
 
-  async getMergeAuthorEmail(commit: GitHubCommitData) {
+  async getMergeAuthorEmail(
+    commit: GitHubCommitData,
+    sessionData: UserWithSiteSessionData
+  ) {
     const {
       author: { name: authorName },
     } = commit
+    const { siteName } = sessionData
+
+    if (!authorName.startsWith("isomergithub")) {
+      // Legacy style of commits, or if the user is not found
+      return this.extractAuthorEmail(commit)
+    }
 
     // Commit was made by our common identity GitHub user
-    if (authorName.startsWith("isomergithub")) {
-      // TODO: Retrieve email address of user from review request table
+    const site = await this.getBySiteName(siteName)
+    if (!site) {
+      return this.extractAuthorEmail(commit)
+    }
```

Comment:
could you go into more detail here? it seems like this conditional is dealing with sites that are not onboarded onto identity right?

Reply:
Yup, I reorganised this part of the code to avoid an arrowhead style of code. But basically, if there are any issues that are encountered along the way, we will fall back to retrieving the email address from the commit itself (i.e. the email address of the git commit author).